src/metabase/models/database.clj

```diff
@@ -88,11 +88,14 @@
   (mi/superuser?))
 
 (defmethod mi/can-write? :model/Database
+  ;; Lack of permission to change database details will also exclude the `details` field from the HTTP response,
+  ;; cf. the implementation of [[metabase.models.interface/to-json]] for `:model/Database`.
   ([instance]
    (mi/can-write? :model/Database (u/the-id instance)))
```

Comment:
Could you change this so that the 1-arity version checks `:is_attached_dwh` on the `instance` directly if it exists, rather than recursively calling into the 2-arity version and then making an unnecessary DB call?

It seems like the 1-arity version of `can-write?` is what's used in most places in `metabase.api.database` so this would be a helpful optimization, especially when we're fetching a large list of DBs.

Reply:
Is https://github.com/metabase/metabase/pull/47247/commits/98ca3b19ed3439d471950252800e28d153a64eae (close to) what you had in mind?

Reply:
Yeah that looks fine. Is `:is_attached_dwh` is added to DBs automatically when they're fetched in the `metabase.api.database` endpoints?

Reply:
Yes, that works automatically. `is_attached_dwh` appears in the HTTP response for `GET /api/database/2`.